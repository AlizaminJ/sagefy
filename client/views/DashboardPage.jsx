import React from 'react'
import { string, shape, arrayOf } from 'prop-types'
import { convertUuidToUuid58 as to58 } from 'uuid58'
import get from 'lodash.get'
import Layout from './components/Layout'
import Icon from './components/Icon'
import Footer from './components/Footer'
import ChooseSubject from './components/ChooseSubject'
import CardSubjectInfo from './components/CardSubjectInfo'
import SubjectForm from './components/SubjectForm'
import CARD_KIND from '../util/card-kind'

export default function DashboardPage({
  role,
  hash,
  name: userName,
  subjects,
  mySubjects,
  myCards,
}) {
  return (
    <Layout hash={hash} page="DashboardPage" title="Dashboard" description="-">
      <section className="my-c">
        <p>
          <em>
            Hi {userName}, welcome back! <Icon i="sagefy" />
          </em>
        </p>
        {/* TODO copy should change based on state */}
        <h1>
          What do you want to learn? <Icon i="dashboard" s="h1" />
        </h1>
      </section>

      {subjects && subjects.length ? (
        <section>
          <h2>
            Choose one of your subjects <Icon i="subject" s="h2" />
          </h2>
          <ChooseSubject subjects={subjects} level="goal" />
          <div className="my-c">
            <h3>
              &hellip;or how about something else? <Icon i="search" s="h3" />
            </h3>
            <form action="/subjects/search">
              <input
                type="search"
                size="40"
                placeholder="example: Music"
                name="q"
              />
              <button type="submit">
                <Icon i="search" /> Search
              </button>
            </form>
          </div>
        </section>
      ) : (
        <section className="ta-c">
          <h2>
            Let&apos;s get you some subjects! <Icon i="search" s="h2" />
          </h2>
          {/* TODO If the user has no subjects, then also list popular subjects here */}
          <form action="/subjects/search">
            <p>
              <input
                type="search"
                size="40"
                placeholder="example: Music"
                autoFocus
                name="q"
              />
            </p>
            <p>
              <button type="submit">
                <Icon i="search" /> Search
              </button>
            </p>
          </form>
        </section>
      )}

      {/*

<section>
  <h2>You've got some new notices ðŸ””</h2>
  <ul className="ls-n">
    <li>
      <p>
        Eileen voted âœ… yes on your proposal on
        <a href="/talk">Music Theory</a>.
      </p>
    </li>
    <li>
      <p>
        Or,
        <strong><a href="/notices">ðŸ”” See all your notices</a>.</strong>
      </p>
    </li>
  </ul>
</section>

*/}

      <section>
        <h2>
          Help us build Sagefy <Icon i="build" s="h2" />
        </h2>

        {!!mySubjects.length && (
          <details>
            <summary>
              <h3 className="d-ib">
                Add some cards <Icon i="card" s="h3" />
              </h3>
            </summary>
            <ul>
              {mySubjects.map(({ entityId, name, cardCount, childCount }) => (
                <li>
                  <a href={`/cards/create?subjectId=${to58(entityId)}`}>
                    Add cards to <q>{name}</q>
                  </a>{' '}
                  <small>
                    ({cardCount} cards, {childCount} child subjects)
                  </small>
                </li>
              ))}
            </ul>
          </details>
        )}

        <details>
          <summary>
            <h3 className="d-ib">
              You can make a new subject <Icon i="subject" s="h3" />
            </h3>
          </summary>
          <SubjectForm role={role} />
        </details>

        {!!mySubjects.length && (
          <details>
            <summary>
              <h3 className="d-ib">
                Review your subjects <Icon i="subject" s="h3" />
              </h3>
            </summary>
            <ul>
              {mySubjects.map(({ entityId, name }) => (
                <li>
                  <a href={`/subjects/${to58(entityId)}`}>
                    Review <q>{name}</q>
                  </a>
                </li>
              ))}
            </ul>
          </details>
        )}

        {!!myCards.length && (
          <details>
            <summary>
              <h3 className="d-ib">
                Review your cards <Icon i="card" s="h3" />
              </h3>
            </summary>
            <ul>
              {myCards.map(({ entityId, name, kind }) => (
                <li>
                  <a
                    href={`/${get(CARD_KIND, [kind, 'url'])}-cards/${to58(
                      entityId
                    )}`}
                  >
                    Review {get(CARD_KIND, [kind, 'name']).toLowerCase()} card{' '}
                    <q>{name}</q>
                  </a>
                </li>
              ))}
            </ul>
          </details>
        )}

        <details>
          <summary>
            <h3 className="d-ib">
              &hellip;or find something else <Icon i="search" s="h3" />
            </h3>
          </summary>
          <form action="/search">
            <input
              type="search"
              size="40"
              placeholder="example: Music"
              name="q"
            />
            <button type="submit">
              <Icon i="search" /> Search
            </button>
          </form>
        </details>

        <CardSubjectInfo />
      </section>

      <Footer role={role} />
    </Layout>
  )
}

DashboardPage.propTypes = {
  hash: string.isRequired,
  role: string,
  name: string,
  subjects: arrayOf(
    shape({
      entityId: string.isRequired,
      name: string.isRequired,
      body: string.isRequired,
    })
  ),
  mySubjects: arrayOf(
    shape({
      entityId: string.isRequired,
      name: string.isRequired,
    })
  ),
  myCards: arrayOf(
    shape({
      entityId: string.isRequired,
      name: string.isRequired,
      kind: string.isRequired,
    })
  ),
}

DashboardPage.defaultProps = {
  role: 'sg_anonymous',
  name: 'friend',
  subjects: [],
  mySubjects: [],
  myCards: [],
}
