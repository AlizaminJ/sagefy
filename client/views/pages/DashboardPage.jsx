import React from 'react'
// import { Link } from 'react-router-dom'
import { string } from 'prop-types'
import Icon from '../components/Icon'
import Footer from '../components/Footer'

export default function DashboardPage({ role }) {
  return (
    <div className="DashboardPage">
      <section>
        <h1>
          Welcome to Sagefy <Icon i="dashboard" s="xxl" />
        </h1>
        <p>
          I&apos;ve reserved your user name and password! That said, there
          isn&apos;t much to do right now.
        </p>
      </section>

      <Footer role={role} />
    </div>
  )
}

DashboardPage.propTypes = {
  role: string,
}

DashboardPage.defaultProps = {
  role: 'sg_anonymous',
}

/*
      <section className="collapse-margins">
        <p>
          <em>Hi Doris, welcome back!</em>
        </p>{' '}
        {/* TODO copy should change based on state *}
        <h1>
          What do you want to learn?{' '}
          <img src="/astrolabe.svg" height="32" alt="astrolabe" />
        </h1>
      </section>

      TODO If the user has no subjects, then list popular subjects here instead
      TODO implement
      <section>
        <h2>
          Choose one of your subjects <Icon i="subject" s="xl" />
        </h2>
        <table>
          <tr>
            <td className="text-align-center collapse-margins">
              <p>
                <button type="submit">
                  <Icon i="up" />
                </button>
              </p>
              <code>15%</code>
            </td>
            <td className="collapse-margins">
              <h3>
                <mark>
                  <Link to="/choose-next">
                    An Introduction to Electronic Music
                  </Link>
                </mark>
              </h3>
              <p>
                A small taste of the basics of electronic music. Learn the
                concepts behind creating and modifying sounds in an electronic
                music system. Learn the ideas behind the tools and systems we
                use to create electronic music.
              </p>
            </td>
          </tr>
          <tr>
            <td className="text-align-center collapse-margins">
              <p>
                <button type="button">
                  <Icon i="up" />
                </button>
              </p>
              <code>0%</code>
            </td>
            <td className="collapse-margins">
              <h3>
                <Link to="/choose-next">Lets Play Classical Guitar</Link>
              </h3>
              <p>
                An introduction to classical guitar. Lets learn some chords. And
                how to read guitar tabulature.
              </p>
            </td>
          </tr>
          <tr>
            <td className="collapse-margins" colSpan="2">
              <h3>
                ...or how about something else? <Icon i="search" s="l" />
              </h3>
              <form action="/search-subjects">
                <input type="search" size="40" placeholder="example: Music" />
                <button type="submit">
                  <Icon i="search" /> Search
                </button>
              </form>
            </td>
          </tr>
        </table>
      </section>


<section>
  <h2>You've got some new notices 🔔</h2>
  <ul className="list-style-none">
    <li>
      <p>
        Eileen voted ✅ yes on your proposal on
        <a href="/talk">Music Theory</a>.
      </p>
    </li>
    <li>
      <p>
        Or,
        <strong><a href="/notices">🔔 See all your notices</a>.</strong>
      </p>
    </li>
  </ul>
</section>

<section>
  <h2>Help us build Sagefy 🌳</h2>

  <details>
    <summary><h3 className="display-inline-block">Add some cards 🃏</h3></summary>
    <ul>
      <li>
        <a href="/create-card"
          >Add cards to 🎧 <em>An Introduction to Electronic Music</em></a
        >
        <small>(0 cards)</small>
      </li>
      <li>
        <a href="/create-card"
          >Add cards to 🎸 <em>Let's Play Classical Guitar</em></a
        >
        <small>(10 cards)</small>
      </li>
      {/* prioritized queue, descend children of followed subjects *}
    </ul>
  </details>

  <details>
    <summary
      ><h3 className="display-inline-block">
        You can make a new subject 💡
      </h3></summary
    >

    <form action="/mocks/create-subject">
      <p>
        <label for="name">What should we call this new subject?</label>
        <input
          type="text"
          value="Music"
          placeholder="example: Introduction to Classical Guitar"
          size="40"
          id="name"
          name="name"
          autofocus
        />
      </p>

      <p>
        <label for="body">What are the goals of this subject?</label>
        <textarea
          placeholder="example: An introduction to classical guitar. Let's learn some chords. And how to read guitar tablature."
          cols="40"
          rows="4"
          id="body"
          name="body"
        ></textarea>
      </p>

      <p>
        <button type="submit">📚 Create Subject</button>
      </p>
    </form>
  </details>

  <details>
    <summary
      ><h3 className="display-inline-block">
        ...or find something else 🕵🏻‍♀️
      </h3></summary
    >
    <form action="/mocks/search">
      <input type="search" size="40" placeholder="example: Music" />
      <button type="submit">🕵🏻‍♀️ Search</button>
    </form>
  </details>
</section>

<section>
  <details>
    <summary>
      <span>🤷🏽‍♀️ What are cards &amp; subjects? 🤷🏽‍♀️</span>
    </summary>
    <ul>
      <li>
        <p>A <strong>card</strong> is a single learning activity. 🃏</p>
        <blockquote>
          <p>Examples: a 3-minute video or a multiple choice question.</p>
        </blockquote>
      </li>
      <li>
        <p>
          A <strong>subject</strong> is a collection of cards and other
          subjects. 📚
        </p>
        <blockquote>
          <p>
            Like a course, but at any scale. Such as “Measures of Central
            Tendency”, “Intro to Statistics”, or even a complete statistics
            program.
          </p>
        </blockquote>
      </li>
    </ul>
  </details>
</section>
*/
