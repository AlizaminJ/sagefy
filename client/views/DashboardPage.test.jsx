import DashboardPage from './DashboardPage'

describe('DashboardPage', () => {
  it('should render the dashboard page', () => {
    expect(
      DashboardPage({ subjects: [], mySubjects: [], myCards: [] })
    ).toMatchSnapshot()
  })
})
