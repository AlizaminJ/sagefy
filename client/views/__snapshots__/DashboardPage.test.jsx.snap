// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPage should render the dashboard page 1`] = `
<Layout
  canonical={null}
  description="-"
  image={null}
  page="DashboardPage"
  title="Dashboard"
>
  <section
    className="my-c"
  >
    <p>
      <em>
        Hi 
        , welcome back! 
        <Icon
          i="sagefy"
          s="m"
        />
      </em>
    </p>
    <h1>
      What do you want to learn? 
      <Icon
        i="dashboard"
        s="h1"
      />
    </h1>
  </section>
  <section
    className="ta-c"
  >
    <h2>
      Let's get you some subjects! 
      <Icon
        i="search"
        s="h2"
      />
    </h2>
    <form
      action="/subjects/search"
    >
      <p>
        <input
          autoFocus={true}
          name="q"
          placeholder="example: Music"
          size="40"
          type="search"
        />
      </p>
      <p>
        <button
          type="submit"
        >
          <Icon
            i="search"
            s="m"
          />
           Search
        </button>
      </p>
    </form>
  </section>
  <section>
    <h2>
      Help us build Sagefy 
      <Icon
        i="cheer"
        s="h2"
      />
    </h2>
    <details>
      <summary>
        <h3
          className="d-ib"
        >
          You can make a new subject 
          <Icon
            i="subject"
            s="h3"
          />
        </h3>
      </summary>
      <CreateSubject
        body={
          Object {
            "body": "",
            "name": "",
          }
        }
        gqlErrors={Object {}}
        name=""
      />
    </details>
    <details>
      <summary>
        <h3
          className="d-ib"
        >
          â€¦or find something else 
          <Icon
            i="search"
            s="h3"
          />
        </h3>
      </summary>
      <form
        action="/search"
      >
        <input
          name="q"
          placeholder="example: Music"
          size="40"
          type="search"
        />
        <button
          type="submit"
        >
          <Icon
            i="search"
            s="m"
          />
           Search
        </button>
      </form>
    </details>
    <CardSubjectInfo />
  </section>
  <Footer
    role="sg_anonymous"
  />
</Layout>
`;
